tornado/iostream.py

```diff
@@ -381,12 +383,15 @@ def write(self, data, callback=None):
             if (self.max_write_buffer_size is not None and
                     self._write_buffer_size + len(data) > self.max_write_buffer_size):
                 raise StreamBufferFullError("Reached maximum write buffer size")
-            # Break up large contiguous strings before inserting them in the
-            # write buffer, so we don't have to recopy the entire thing
-            # as we slice off pieces to send to the socket.
-            WRITE_BUFFER_CHUNK_SIZE = 128 * 1024
-            for i in range(0, len(data), WRITE_BUFFER_CHUNK_SIZE):
-                self._write_buffer.append(data[i:i + WRITE_BUFFER_CHUNK_SIZE])
+            if WINDOWS:
+                # Break up large contiguous strings before inserting them in the
+                # write buffer, so we don't have to recopy the entire thing
+                # as we slice off pieces to send to the socket.
+                WRITE_BUFFER_CHUNK_SIZE = 128 * 1024
+                for i in range(0, len(data), WRITE_BUFFER_CHUNK_SIZE):
+                    self._write_buffer.append(data[i:i + WRITE_BUFFER_CHUNK_SIZE])
```

Comment:
Does `socket.send` on windows support memoryview objects? If so, we may be able to avoid this splitting and just give `send()` a memoryview.